rmf_task/include/rmf_task/requests/factory/ChargeBatteryFactory.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef RMF_TASK__REQUESTS__FACTORY__CHARGEBATTERYFACTORY_HPP
+#define RMF_TASK__REQUESTS__FACTORY__CHARGEBATTERYFACTORY_HPP
+
+#include <rmf_task/RequestFactory.hpp>
+#include <rmf_task/agv/State.hpp>
+
+#include <rmf_utils/impl_ptr.hpp>
+
+namespace rmf_task {
+namespace requests {
+
+//==============================================================================
+class ChargeBatteryFactory : public RequestFactory
```

Comment:
I wonder if we should rename `ReturnToChargerFactory` to something like `ParkRobotFactory` with a `ParkRobot` task. For robots whose only parking space is their charger, this will naturally make them return to their charger. In the future when we have the parking reservation system, we can evolve the implementation to have idle robots park in their nearest available parking spot. Although the `ParkRobotFactory` should probably still send the robots back to get charged if their battery is low enough. Otherwise its battery level might dwindle too low before receiving a new task, potentially leaving it stranded in its parking spot and requiring manual intervention to get it going again.

Reply:
I'm okay with renaming to `ParkRobotFactory` and implementing a new task. But should the new task be called `Station` instead? This would make its integration in downstream fleet adapters a bit easier since most of the task messages already have a `station` field 
https://github.com/open-rmf/rmf_internal_msgs/blob/36f15f4dcd8d8ea4ff41f7e0a14f9ee1e7b1bff7/rmf_task_msgs/msg/TaskType.msg#L2
 https://github.com/open-rmf/rmf_internal_msgs/blob/36f15f4dcd8d8ea4ff41f7e0a14f9ee1e7b1bff7/rmf_task_msgs/msg/TaskDescription.msg#L11
 
 Or is it okay to call it `ParkRobot` and just have downstream adapters report it as a `station` task?

Reply:
When we migrate to the [more flexible task system](https://github.com/open-rmf/rmf_ros2/issues/107) all of our message definitions are going to be blasted away, so I don't think there's much point in constraining ourselves based on old definitions.

That being said, I think "Station" is a bit different from "Park". I believe "Station" is more closely related to the `ResponsiveWait` behavior, where the robot might be positioned anywhere and its job is to keep itself there, perhaps while still giving way to other traffic. Whereas "Park" means the robot has reserved an exclusive parking spot for itself and should sit on that parking spot without fear of being disrupted.

But also the term "station" predates any of the conventions and terminologies that we're using right now but has also gone unused this entire time, so I suppose its exact meaning is still up to interpretation. In any case, we should flesh out how exactly to use that word in our upcoming conversations on the task system, and we shouldn't tie ourselves down to the current message definitions since they need to drastically change anyway.

Reply:
Renamed to `ParkRobotFactory` https://github.com/open-rmf/rmf_task/pull/28/commits/0353e21ec91a0e15dc04259ff17433c96f8aea73

The implementation accepts an optional `parking_waypoint`. If `nullopt` is supplied, the `charging_waypoint` is used.

The factory still return a Loop request. I wasn't sure if writing a new `ParkRobot` request made sense given the migration to the flexible task system in the near future. The internal implementation would have been the same as `Loop` where num_loops = 1. Also this way, we wouldn't need to change much on the fleet adapter side. But if you feel it would be better to have that `ParkRobot` request, I'm totally okay with implementing it 😄